pkg/cloud/request.go

```diff
@@ -0,0 +1,196 @@
+package cloud
+
+import (
+	"context"
+	"github.com/applike/gosoline/pkg/mon"
+	"github.com/cenkalti/backoff"
+	"time"
+)
+
+const (
+	metricNameErrorCount = "AwsRequestErrorCount"
+	metricNameRetryCount = "AwsRequestRetryCount"
+)
+
+type delayedCancelContext struct {
+	context.Context
+	cancel context.CancelFunc
+	stop   chan struct{}
+}
+
+func (c *delayedCancelContext) Stop() {
+	close(c.stop)
+}
+
+func WithDelayedCancelContext(parentCtx context.Context, delay time.Duration) *delayedCancelContext {
+	stop := make(chan struct{})
+	delayedCtx, cancel := context.WithCancel(context.Background())
```

Comment:
Do we want to carry over any tracing or other things from the parent context?

Reply:
if we want to do that, i would want to do that in a follow up pr..
currently it will not hurt that much, as we're not doing any logging in the AwsRequestFunction